src/analysis/retail/evoker/shared/modules/talents/hero/chronowarden/DoubleTime.tsx

```diff
@@ -0,0 +1,129 @@
+import SPELLS from 'common/SPELLS';
+import { TALENTS_EVOKER } from 'common/TALENTS';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, {
+  ApplyBuffEvent,
+  ApplyDebuffEvent,
+  DamageEvent,
+  EmpowerEndEvent,
+  GetRelatedEvent,
+  GetRelatedEvents,
+  RefreshBuffEvent,
+  RefreshDebuffEvent,
+} from 'parser/core/Events';
+import { Options } from 'parser/core/EventSubscriber';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import TalentSpellText from 'parser/ui/TalentSpellText';
+import { SpellLink } from 'interface';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import {
+  getDreamBreathCast,
+  getDreamBreathHealing,
+} from 'analysis/retail/evoker/preservation/normalizers/EventLinking/helpers';
+import HIT_TYPES from 'game/HIT_TYPES';
+import {
+  FIRE_BREATH,
+  FIRE_BREATH_CAST,
+} from 'analysis/retail/evoker/preservation/normalizers/EventLinking/constants';
+
+class DoubleTime extends Analyzer {
+  //Dreambreath
+  totalDbs: number = 0;
+  sumDbDurations: number = 0;
+  dbDurations: number[] = [0, 16, 12, 8, 4];
+  //Firebreath
+  blastFurnace: boolean = false;
+  totalFbs: number = 0;
+  sumFbDurations: number = 0;
+  fbDurations: number[] = [0, 20, 14, 8, 2];
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(TALENTS_EVOKER.DOUBLE_TIME_TALENT);
+    this.blastFurnace = this.selectedCombatant.hasTalent(TALENTS_EVOKER.BLAST_FURNACE_TALENT);
+
+    //Dream Breath
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell([SPELLS.DREAM_BREATH, SPELLS.DREAM_BREATH_ECHO]),
+      this.onApplyDb,
+    );
+    this.addEventListener(
+      Events.refreshbuff.by(SELECTED_PLAYER).spell([SPELLS.DREAM_BREATH, SPELLS.DREAM_BREATH_ECHO]),
+      this.onApplyDb,
+    );
+
+    //Fire Breath
+    this.addEventListener(
+      Events.applydebuff.by(SELECTED_PLAYER).spell(SPELLS.FIRE_BREATH_DOT),
+      this.onApplyFb,
+    );
+    this.addEventListener(
+      Events.refreshdebuff.by(SELECTED_PLAYER).spell(SPELLS.FIRE_BREATH_DOT),
+      this.onApplyFb,
+    );
+  }
+
+  onApplyDb(event: ApplyBuffEvent | RefreshBuffEvent) {
+    const castEvent = getDreamBreathCast(event);
+    if (castEvent) {
+      this.totalDbs += 1;
+      const healEvents = getDreamBreathHealing(event);
+      let extensions = 0;
+      for (const heal of healEvents) {
+        if (heal.hitType === HIT_TYPES.CRIT) {
+          extensions += 2;
```

Comment:
Can you move this 2 into a global constant called `DOUBLE_TIME_EXTENSION` so the meaning of the 2 is apparent, also makes it easier for refactoring if they change the number